File: php/stack_program.js
class Stack {
  constructor(){ this.items = []; }
  push(x){ this.items.push(x); }
  pop(){ return this.items.length ? this.items.pop() : "Underflow"; }
  peek(){ return this.items.length ? this.items[this.items.length - 1] : "Empty"; }
  display(){ console.log(this.items); }
}
let s = new Stack();
s.push(10); s.push(20); s.push(30);
s.display(); console.log("Top:", s.peek());
console.log("Popped:", s.pop()); s.display();

File: c++/stack_program.py
class Stack:
    def __init__(self): self.items = []
    def push(self, item): self.items.append(item)
    def pop(self): return self.items.pop() if self.items else "Underflow"
    def peek(self): return self.items[-1] if self.items else "Empty"
    def display(self): print(self.items)

s = Stack()
s.push(10); s.push(20); s.push(30)
s.display()
print("Top:", s.peek())
print("Popped:", s.pop())
s.display()
